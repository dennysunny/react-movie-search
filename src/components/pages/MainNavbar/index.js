import withConfig from '../../queries/withConfig';
import MainNavbar from '../../templates/MainNavbar';

export default withConfig()(MainNavbar);
