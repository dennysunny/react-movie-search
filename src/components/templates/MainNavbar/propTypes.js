import { object } from 'prop-types';

export const propTypes = {
  config: object.isRequired,
  history: object.isRequired,
};

export const defaultProps = {
};
