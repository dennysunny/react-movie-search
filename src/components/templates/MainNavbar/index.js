/* eslint-disable camelcase */
import React, { useState } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import { Link } from 'react-router-dom';

import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import MenuItem from '@material-ui/core/MenuItem';
import MenuList from '@material-ui/core/MenuList';
import Menu from '@material-ui/core/Menu';
import MenuIcon from '@material-ui/icons/Menu';
import IconButton from '@material-ui/core/IconButton';
import AccountCircle from '@material-ui/icons/AccountCircle';
import Drawer from '@material-ui/core/Drawer';
import withWidth, { isWidthUp } from '@material-ui/core/withWidth';

import { propTypes, isDrawerWrapperPropTypes } from './propTypes';
import { FirebaseAuthContext } from '../../../firebase/FirebaseAuthProvider';
import SignOutButton from '../../atoms/SignOut';
import NavSearch from '../../atoms/NavSearch';

const useStyles = makeStyles(theme => ({
  appBar: {
    background: '#1f1d1d',
    maxHeight: '63px',
  },
  logo: {
    marginRight: theme.spacing(2),
    whiteSpace: 'nowrap',
  },
  menuButton: {
    marginRight: theme.spacing(2),
    [theme.breakpoints.up('md')]: {
      display: 'none',
    },
  },
  menuList: {
    display: 'flex',
    flexGrow: 1,
  },
  menuListUser: {
    display: 'flex',
  },
  menuItem: {
    color: '#fff',
    textDecoration: 'none',
  },
  searchMenu: {
    overflow: 'visible',
  },
  drawerPaper: {
    background: '#000',

    '& $menuList': {
      flexDirection: 'column',
      marginTop: theme.spacing(3),

      '& $menuItem': {
        padding: '0 16px',
        width: '100%',
      },
    },
  },
}));

const MainNavbar = (props) => {
  const classes = useStyles();
  const { config, width } = props;
  const [anchorEl, setAnchorEl] = useState(null);
  const [mobileOpen, setMobileOpen] = useState(false);

  const MainMenu = () => (
    <MenuList className={classes.menuList}>
      <MenuItem>
        <Link to="/movies" className={classes.menuItem}>
          <Typography variant="button">
            MOVIES
          </Typography>
        </Link>
      </MenuItem>
      <MenuItem>
        <Link to="/tvs" className={classes.menuItem}>
          <Typography variant="button">
            TVS
          </Typography>
        </Link>
      </MenuItem>
      <MenuItem>
        <Link to="/person" className={classes.menuItem}>
          <Typography variant="button">
            PEOPLE
          </Typography>
        </Link>
      </MenuItem>
      <MenuItem>
        <Link to="/watchlist" className={classes.menuItem}>
          <Typography variant="button">
            WATCHLIST&nbsp;
            <span className="fa fa-bookmark" />
          </Typography>
        </Link>
      </MenuItem>
    </MenuList>
  );

  const UserMenu = () => (
    <FirebaseAuthContext.Consumer>
      {({ user, authUser }) => ((!authUser)
        ? (
          <MenuList className={classes.menuListUser}>
            <MenuItem>
              <Link to="/login" className={classes.menuItem}>
                <Typography variant="body2">
                  Login
                </Typography>
              </Link>
            </MenuItem>
            <MenuItem>
              <Link to="/signup" className={classes.menuItem}>
                <Typography variant="body2">
                  Sign up
                </Typography>
              </Link>
            </MenuItem>
          </MenuList>
        )
        : (
          <div>
            <IconButton
              aria-label="Account of current user"
              aria-controls="menu-appbar"
              aria-haspopup="true"
              onClick={e => setAnchorEl(e.currentTarget)}
              color="inherit"
            >
              <AccountCircle />
            </IconButton>

            <Menu
              id="menu-appbar"
              anchorEl={anchorEl}
              anchorOrigin={{
                vertical: 'top',
                horizontal: 'right',
              }}
              keepMounted
              transformOrigin={{
                vertical: 'top',
                horizontal: 'right',
              }}
              open={Boolean(anchorEl)}
              onClose={() => setAnchorEl(null)}
            >
              <MenuItem disabled>
                {user ? user.username : null}
              </MenuItem>
              <MenuItem href="#" disabled>
                view profile
              </MenuItem>
              <MenuItem divider />
              <MenuItem>
                <Link to="/watchlist">
                  Watchlist
                </Link>
              </MenuItem>
              <MenuItem divider />
              <MenuItem>
                <SignOutButton />
              </MenuItem>
            </Menu>
          </div>
        ))}
    </FirebaseAuthContext.Consumer>
  );

  const handleDrawerToggle = () => {
    setMobileOpen(!mobileOpen);
  };

  const IsDrawerWrapper = ({ children }) => (
    isWidthUp('md', width)
      ? children
      : (
        <Drawer
          // container={container}
          variant="temporary"
          anchor="left"
          open={mobileOpen}
          onClose={handleDrawerToggle}
          classes={{
            paper: classes.drawerPaper,
          }}
          ModalProps={{
            keepMounted: true,
          }}
        >
          {children}
        </Drawer>
      ));

  IsDrawerWrapper.propTypes = isDrawerWrapperPropTypes;

  return (
    <AppBar color="primary" className={classes.appBar}>
      <Toolbar variant="dense">
        <IconButton
          color="inherit"
          aria-label="Open drawer"
          edge="start"
          onClick={handleDrawerToggle}
          className={classes.menuButton}
        >
          <MenuIcon />
        </IconButton>
        <Link to="/" className={classes.menuItem}>
          <Typography variant="h6" className={classes.logo}>
            Movie Search
          </Typography>
        </Link>
        <IsDrawerWrapper>
          <MainMenu />
        </IsDrawerWrapper>
        <div style={{ flexGrow: 1 }} />
        <NavSearch config={config} />
        <UserMenu />
      </Toolbar>
    </AppBar>
  );
};

export default withWidth()(MainNavbar);

MainNavbar.propTypes = propTypes;
