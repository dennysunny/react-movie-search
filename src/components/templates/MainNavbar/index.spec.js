import React from 'react';
import { shallow, mount } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';

import MainNavbar from '.';

const mockProps = {
  config: {
    images: {
      secure_base_url: 'path/to/image',
      profile_sizes: ['1', '2', '3', '4'],
      poster_sizes: ['1', '2', '3', '4'],
      backdrop_sizes: ['1', '2', '3', '4'],
    },
  },
  history: new BrowserRouter().history,
};

describe('Templates/MainNavbar', () => {
  it('should render as expected', () => {
    const wrapper = shallow(<MainNavbar.WrappedComponent {...mockProps} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should handle search options', () => {
    const wrapper = shallow(<MainNavbar.WrappedComponent {...mockProps} />);
    jest.spyOn(wrapper.instance(), 'handleSearch')
      .mockImplementation(() => ({
        options: [
          {
            id: 1,
            title: 'title',
            release_date: '01.01.2010',
            poster_path: 'path/to/image',
          },
        ],
        isLoading: false,
      }));

    wrapper.instance().handleSearch('test');

    expect(wrapper.instance().handleSearch).toHaveBeenCalled();
  });

  it('should handle select options', () => {
    const wrapper = mount(<BrowserRouter><MainNavbar {...mockProps} /></BrowserRouter>);

    wrapper.find(AsyncTypeahead).simulate('change', [{ media_type: 'movie', id: '1' }]);
    expect(wrapper.props().children.props.history.location.pathname).toEqual('/');
  });
});
