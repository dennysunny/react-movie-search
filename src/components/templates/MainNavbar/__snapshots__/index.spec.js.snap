// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Templates/MainNavbar should render as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MainNavbar
    config={
      Object {
        "images": Object {
          "backdrop_sizes": Array [
            "1",
            "2",
            "3",
            "4",
          ],
          "poster_sizes": Array [
            "1",
            "2",
            "3",
            "4",
          ],
          "profile_sizes": Array [
            "1",
            "2",
            "3",
            "4",
          ],
          "secure_base_url": "path/to/image",
        },
      }
    }
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "bsStyle": "default",
      "children": Array [
        <NavbarHeader>
          <NavbarBrand>
            <Link
              replace={false}
              to="/"
            >
              Movie Search
            </Link>
          </NavbarBrand>
          <NavbarToggle />
        </NavbarHeader>,
        <NavbarCollapse>
          <NavbarText
            componentClass="p"
            pullLeft={false}
            pullRight={false}
          >
            <Link
              replace={false}
              to="/movies"
            >
              MOVIES
            </Link>
          </NavbarText>
          <NavbarText
            componentClass="p"
            pullLeft={false}
            pullRight={false}
          >
            <Link
              replace={false}
              to="/tvs"
            >
              TVS
            </Link>
          </NavbarText>
          <NavbarText
            componentClass="p"
            pullLeft={false}
            pullRight={false}
          >
            <Link
              replace={false}
              to="/person"
            >
              PEOPLE
            </Link>
          </NavbarText>
          <NavbarText
            componentClass="p"
            pullLeft={false}
            pullRight={false}
          >
            <Link
              replace={false}
              to="/watchlist"
            >
              WATCHLIST 
              <span
                className="fa fa-bookmark"
              />
            </Link>
          </NavbarText>
          <NavbarForm
            componentClass="div"
            pullLeft={true}
            pullRight={false}
          >
            <AsyncContainer(TypeaheadContainer(WrappedTypeahead))
              align="justify"
              delay={200}
              isLoading={false}
              labelKey={[Function]}
              minLength={2}
              onChange={[Function]}
              onSearch={[Function]}
              options={Array []}
              placeholder="Search for a movie or a person..."
              promptText="Type to search..."
              renderMenuItemChildren={[Function]}
              searchText="Searching..."
              useCache={true}
            />
          </NavbarForm>
          <Nav
            bsClass="nav"
            id="nav-user"
            justified={false}
            pullLeft={false}
            pullRight={true}
            stacked={false}
          >
            <NavItem
              active={false}
              disabled={false}
              eventKey="login"
              href="/login"
            >
              Login
            </NavItem>
            <NavItem
              active={false}
              disabled={false}
              eventKey="signup"
              href="/signup"
            >
              Sign up
            </NavItem>
          </Nav>
        </NavbarCollapse>,
      ],
      "fixedTop": true,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <NavbarBrand>
              <Link
                replace={false}
                to="/"
              >
                Movie Search
              </Link>
            </NavbarBrand>,
            <NavbarToggle />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Link
                replace={false}
                to="/"
              >
                Movie Search
              </Link>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Movie Search",
                "replace": false,
                "to": "/",
              },
              "ref": null,
              "rendered": "Movie Search",
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <NavbarText
              componentClass="p"
              pullLeft={false}
              pullRight={false}
            >
              <Link
                replace={false}
                to="/movies"
              >
                MOVIES
              </Link>
            </NavbarText>,
            <NavbarText
              componentClass="p"
              pullLeft={false}
              pullRight={false}
            >
              <Link
                replace={false}
                to="/tvs"
              >
                TVS
              </Link>
            </NavbarText>,
            <NavbarText
              componentClass="p"
              pullLeft={false}
              pullRight={false}
            >
              <Link
                replace={false}
                to="/person"
              >
                PEOPLE
              </Link>
            </NavbarText>,
            <NavbarText
              componentClass="p"
              pullLeft={false}
              pullRight={false}
            >
              <Link
                replace={false}
                to="/watchlist"
              >
                WATCHLIST 
                <span
                  className="fa fa-bookmark"
                />
              </Link>
            </NavbarText>,
            <NavbarForm
              componentClass="div"
              pullLeft={true}
              pullRight={false}
            >
              <AsyncContainer(TypeaheadContainer(WrappedTypeahead))
                align="justify"
                delay={200}
                isLoading={false}
                labelKey={[Function]}
                minLength={2}
                onChange={[Function]}
                onSearch={[Function]}
                options={Array []}
                placeholder="Search for a movie or a person..."
                promptText="Type to search..."
                renderMenuItemChildren={[Function]}
                searchText="Searching..."
                useCache={true}
              />
            </NavbarForm>,
            <Nav
              bsClass="nav"
              id="nav-user"
              justified={false}
              pullLeft={false}
              pullRight={true}
              stacked={false}
            >
              <NavItem
                active={false}
                disabled={false}
                eventKey="login"
                href="/login"
              >
                Login
              </NavItem>
              <NavItem
                active={false}
                disabled={false}
                eventKey="signup"
                href="/signup"
              >
                Sign up
              </NavItem>
            </Nav>,
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Link
                replace={false}
                to="/movies"
              >
                MOVIES
              </Link>,
              "componentClass": "p",
              "pullLeft": false,
              "pullRight": false,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "MOVIES",
                "replace": false,
                "to": "/movies",
              },
              "ref": null,
              "rendered": "MOVIES",
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Link
                replace={false}
                to="/tvs"
              >
                TVS
              </Link>,
              "componentClass": "p",
              "pullLeft": false,
              "pullRight": false,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "TVS",
                "replace": false,
                "to": "/tvs",
              },
              "ref": null,
              "rendered": "TVS",
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Link
                replace={false}
                to="/person"
              >
                PEOPLE
              </Link>,
              "componentClass": "p",
              "pullLeft": false,
              "pullRight": false,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "PEOPLE",
                "replace": false,
                "to": "/person",
              },
              "ref": null,
              "rendered": "PEOPLE",
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Link
                replace={false}
                to="/watchlist"
              >
                WATCHLIST 
                <span
                  className="fa fa-bookmark"
                />
              </Link>,
              "componentClass": "p",
              "pullLeft": false,
              "pullRight": false,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  "WATCHLIST ",
                  <span
                    className="fa fa-bookmark"
                  />,
                ],
                "replace": false,
                "to": "/watchlist",
              },
              "ref": null,
              "rendered": Array [
                "WATCHLIST ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "fa fa-bookmark",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "span",
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <AsyncContainer(TypeaheadContainer(WrappedTypeahead))
                align="justify"
                delay={200}
                isLoading={false}
                labelKey={[Function]}
                minLength={2}
                onChange={[Function]}
                onSearch={[Function]}
                options={Array []}
                placeholder="Search for a movie or a person..."
                promptText="Type to search..."
                renderMenuItemChildren={[Function]}
                searchText="Searching..."
                useCache={true}
              />,
              "componentClass": "div",
              "pullLeft": true,
              "pullRight": false,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "align": "justify",
                "delay": 200,
                "isLoading": false,
                "labelKey": [Function],
                "minLength": 2,
                "onChange": [Function],
                "onSearch": [Function],
                "options": Array [],
                "placeholder": "Search for a movie or a person...",
                "promptText": "Type to search...",
                "renderMenuItemChildren": [Function],
                "searchText": "Searching...",
                "useCache": true,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "nav",
              "children": Array [
                <NavItem
                  active={false}
                  disabled={false}
                  eventKey="login"
                  href="/login"
                >
                  Login
                </NavItem>,
                <NavItem
                  active={false}
                  disabled={false}
                  eventKey="signup"
                  href="/signup"
                >
                  Sign up
                </NavItem>,
              ],
              "id": "nav-user",
              "justified": false,
              "pullLeft": false,
              "pullRight": true,
              "stacked": false,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "active": false,
                  "children": "Login",
                  "disabled": false,
                  "eventKey": "login",
                  "href": "/login",
                },
                "ref": null,
                "rendered": "Login",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "active": false,
                  "children": "Sign up",
                  "disabled": false,
                  "eventKey": "signup",
                  "href": "/signup",
                },
                "ref": null,
                "rendered": "Sign up",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          undefined,
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bsStyle": "default",
        "children": Array [
          <NavbarHeader>
            <NavbarBrand>
              <Link
                replace={false}
                to="/"
              >
                Movie Search
              </Link>
            </NavbarBrand>
            <NavbarToggle />
          </NavbarHeader>,
          <NavbarCollapse>
            <NavbarText
              componentClass="p"
              pullLeft={false}
              pullRight={false}
            >
              <Link
                replace={false}
                to="/movies"
              >
                MOVIES
              </Link>
            </NavbarText>
            <NavbarText
              componentClass="p"
              pullLeft={false}
              pullRight={false}
            >
              <Link
                replace={false}
                to="/tvs"
              >
                TVS
              </Link>
            </NavbarText>
            <NavbarText
              componentClass="p"
              pullLeft={false}
              pullRight={false}
            >
              <Link
                replace={false}
                to="/person"
              >
                PEOPLE
              </Link>
            </NavbarText>
            <NavbarText
              componentClass="p"
              pullLeft={false}
              pullRight={false}
            >
              <Link
                replace={false}
                to="/watchlist"
              >
                WATCHLIST 
                <span
                  className="fa fa-bookmark"
                />
              </Link>
            </NavbarText>
            <NavbarForm
              componentClass="div"
              pullLeft={true}
              pullRight={false}
            >
              <AsyncContainer(TypeaheadContainer(WrappedTypeahead))
                align="justify"
                delay={200}
                isLoading={false}
                labelKey={[Function]}
                minLength={2}
                onChange={[Function]}
                onSearch={[Function]}
                options={Array []}
                placeholder="Search for a movie or a person..."
                promptText="Type to search..."
                renderMenuItemChildren={[Function]}
                searchText="Searching..."
                useCache={true}
              />
            </NavbarForm>
            <Nav
              bsClass="nav"
              id="nav-user"
              justified={false}
              pullLeft={false}
              pullRight={true}
              stacked={false}
            >
              <NavItem
                active={false}
                disabled={false}
                eventKey="login"
                href="/login"
              >
                Login
              </NavItem>
              <NavItem
                active={false}
                disabled={false}
                eventKey="signup"
                href="/signup"
              >
                Sign up
              </NavItem>
            </Nav>
          </NavbarCollapse>,
        ],
        "fixedTop": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <NavbarBrand>
                <Link
                  replace={false}
                  to="/"
                >
                  Movie Search
                </Link>
              </NavbarBrand>,
              <NavbarToggle />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Link
                  replace={false}
                  to="/"
                >
                  Movie Search
                </Link>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Movie Search",
                  "replace": false,
                  "to": "/",
                },
                "ref": null,
                "rendered": "Movie Search",
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <NavbarText
                componentClass="p"
                pullLeft={false}
                pullRight={false}
              >
                <Link
                  replace={false}
                  to="/movies"
                >
                  MOVIES
                </Link>
              </NavbarText>,
              <NavbarText
                componentClass="p"
                pullLeft={false}
                pullRight={false}
              >
                <Link
                  replace={false}
                  to="/tvs"
                >
                  TVS
                </Link>
              </NavbarText>,
              <NavbarText
                componentClass="p"
                pullLeft={false}
                pullRight={false}
              >
                <Link
                  replace={false}
                  to="/person"
                >
                  PEOPLE
                </Link>
              </NavbarText>,
              <NavbarText
                componentClass="p"
                pullLeft={false}
                pullRight={false}
              >
                <Link
                  replace={false}
                  to="/watchlist"
                >
                  WATCHLIST 
                  <span
                    className="fa fa-bookmark"
                  />
                </Link>
              </NavbarText>,
              <NavbarForm
                componentClass="div"
                pullLeft={true}
                pullRight={false}
              >
                <AsyncContainer(TypeaheadContainer(WrappedTypeahead))
                  align="justify"
                  delay={200}
                  isLoading={false}
                  labelKey={[Function]}
                  minLength={2}
                  onChange={[Function]}
                  onSearch={[Function]}
                  options={Array []}
                  placeholder="Search for a movie or a person..."
                  promptText="Type to search..."
                  renderMenuItemChildren={[Function]}
                  searchText="Searching..."
                  useCache={true}
                />
              </NavbarForm>,
              <Nav
                bsClass="nav"
                id="nav-user"
                justified={false}
                pullLeft={false}
                pullRight={true}
                stacked={false}
              >
                <NavItem
                  active={false}
                  disabled={false}
                  eventKey="login"
                  href="/login"
                >
                  Login
                </NavItem>
                <NavItem
                  active={false}
                  disabled={false}
                  eventKey="signup"
                  href="/signup"
                >
                  Sign up
                </NavItem>
              </Nav>,
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <Link
                  replace={false}
                  to="/movies"
                >
                  MOVIES
                </Link>,
                "componentClass": "p",
                "pullLeft": false,
                "pullRight": false,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "MOVIES",
                  "replace": false,
                  "to": "/movies",
                },
                "ref": null,
                "rendered": "MOVIES",
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <Link
                  replace={false}
                  to="/tvs"
                >
                  TVS
                </Link>,
                "componentClass": "p",
                "pullLeft": false,
                "pullRight": false,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "TVS",
                  "replace": false,
                  "to": "/tvs",
                },
                "ref": null,
                "rendered": "TVS",
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <Link
                  replace={false}
                  to="/person"
                >
                  PEOPLE
                </Link>,
                "componentClass": "p",
                "pullLeft": false,
                "pullRight": false,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "PEOPLE",
                  "replace": false,
                  "to": "/person",
                },
                "ref": null,
                "rendered": "PEOPLE",
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <Link
                  replace={false}
                  to="/watchlist"
                >
                  WATCHLIST 
                  <span
                    className="fa fa-bookmark"
                  />
                </Link>,
                "componentClass": "p",
                "pullLeft": false,
                "pullRight": false,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    "WATCHLIST ",
                    <span
                      className="fa fa-bookmark"
                    />,
                  ],
                  "replace": false,
                  "to": "/watchlist",
                },
                "ref": null,
                "rendered": Array [
                  "WATCHLIST ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "fa fa-bookmark",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "span",
                  },
                ],
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <AsyncContainer(TypeaheadContainer(WrappedTypeahead))
                  align="justify"
                  delay={200}
                  isLoading={false}
                  labelKey={[Function]}
                  minLength={2}
                  onChange={[Function]}
                  onSearch={[Function]}
                  options={Array []}
                  placeholder="Search for a movie or a person..."
                  promptText="Type to search..."
                  renderMenuItemChildren={[Function]}
                  searchText="Searching..."
                  useCache={true}
                />,
                "componentClass": "div",
                "pullLeft": true,
                "pullRight": false,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "align": "justify",
                  "delay": 200,
                  "isLoading": false,
                  "labelKey": [Function],
                  "minLength": 2,
                  "onChange": [Function],
                  "onSearch": [Function],
                  "options": Array [],
                  "placeholder": "Search for a movie or a person...",
                  "promptText": "Type to search...",
                  "renderMenuItemChildren": [Function],
                  "searchText": "Searching...",
                  "useCache": true,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "nav",
                "children": Array [
                  <NavItem
                    active={false}
                    disabled={false}
                    eventKey="login"
                    href="/login"
                  >
                    Login
                  </NavItem>,
                  <NavItem
                    active={false}
                    disabled={false}
                    eventKey="signup"
                    href="/signup"
                  >
                    Sign up
                  </NavItem>,
                ],
                "id": "nav-user",
                "justified": false,
                "pullLeft": false,
                "pullRight": true,
                "stacked": false,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "active": false,
                    "children": "Login",
                    "disabled": false,
                    "eventKey": "login",
                    "href": "/login",
                  },
                  "ref": null,
                  "rendered": "Login",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "active": false,
                    "children": "Sign up",
                    "disabled": false,
                    "eventKey": "signup",
                    "href": "/signup",
                  },
                  "ref": null,
                  "rendered": "Sign up",
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            undefined,
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
